lightning/src/util/ser_macros.rs

```diff
@@ -1065,6 +1065,10 @@ macro_rules! impl_writeable_tlv_based_enum {
 /// when [`MaybeReadable`] is practical instead of just [`Readable`] as it provides an upgrade path for
 /// new variants to be added which are simply ignored by existing clients.
 ///
+/// Note that variants written as `$tuple_variant`s will not support downgrading no matter the
+/// `$tuple_variant_id` used, however any variants written as a normal variant will support
```

Comment:
Maybe `$variant_id` rather than "variants written as a normal variant"? That sentence is a bit of a mouthful with the number of times "variant" is used.